nav2_common/cmake/nav2_package.cmake

```diff
@@ -32,6 +32,12 @@ macro(nav2_package)
     set(CMAKE_CXX_STANDARD 14)
   endif()
 
+  if(CMAKE_CXX_STANDARD LESS 14)
```

Comment:
Delete this, nothing will even work without C++14, which is why its set above. If someone really wants to override it, that's fine, but then they're probably not using our macros

Reply:
"nothing will even work without C++14" That's why I have added this.

in the code 
```
if(NOT CMAKE_CXX_STANDARD)
  set(CMAKE_CXX_STANDARD 14)
endif()
```

the standard will be set if it is not previously defined, now say some other package raised a usage requirement and that is < c++11 which will be set by its own config and that will end up raising a tonne of compiler errors crying about many undefined functions and other utilities. 

Here is a relevant example of vtk(dependency of PCL) config 
```
set(VTK_USE_CXX11_FEATURES OFF)
if (VTK_USE_CXX11_FEATURES)
  if (CMAKE_VERSION VERSION_LESS "3.1")
    message(FATAL_ERROR "VTK was built with C++11 features using CMake's CXX_STANDARD_REQUIRED feature.  Configuring projects this way requires a newer version of CMake.  Please update to CMake 3.1 or newer")
  endif()
  set(CMAKE_CXX_STANDARD 11)
  set(CMAKE_CXX_STANDARD_REQUIRED 11)
endif()
 ``` 

this code can change these settings(if on) and then the error will be unidentifiable so I think it is better if we perceive it in advance and stop further processing.  

Even if somebody wants to override the default standard, it should be >=14. that's why it should be fine if we restrict the users/developers to std >=14.